ConsoleProject/src/view/Frontend.cs

```diff
@@ -125,13 +126,34 @@ public class Frontend {
             };
             return listBox;
         }
+        
+        private void ParseNewFile(string[] fileList) {
+            if (fileList.Length > 0) { // is this IF required?
+                try {
+                    images = new TextureConverter().ParseFile(fileList[0]); // TODO only reads first file
+                    this.LoadImages(fileList[0]);
+                } catch (InvalidDataException ex) {
+                    Console.WriteLine(ex);
+                    MessageBox.Show("No images found!\nIncorrect file?");
+                }
+            }
+        }
 
         private void DrawPicture(MouseEventArgs mouseEvent = null) {
+            if (obmp == null) return;
```

Comment:
Add missing brackets. Also write an info log to the console, why is this an error. If it's an error maybe throwing an exception is better. I haven't yet understood the connection between bmp and obmp.

Reply:
I think this happens when no XET (and thus bitmap) has been loaded.
Example: You start the EXE without providing a file, so the form shows, but no default image is loaded.
If at this point you do not return from this function, you get an exception when trying to zoom in or out.

As for bmp vs obmp, bmp is the currently displayed image (including zoom level) while obmp is the original image (default zoom) which allows for the ratio calculation.